### PR TITLE
Generalize Cloudy

### PR DESCRIPTION
**Pull request by edejong-caltech**
Generalizes the implementation of Cloudy to use any number of moments

### CODE REVIEW

File: test/create_parameters.jl [hunk 1]
```diff
@@ -85,14 +85,57 @@ function create_kid_parameters(toml_dict)
     return kid_params
 end
 
+function determine_cloudy_disttypes(NM::Int, default_gamma=true)
```
Comment (sajjadazimi) [inline test/create_parameters.jl:88]: When default is gamma I expect the first distribution to be gamma. Right now there is no way to make the first dist gamma and the rest exp. While sometimes cloud is assumed gamma and rain exp.

**Review by sajjadazimi — CHANGES_REQUESTED** [review]
Thanks for adding these features. Overall looks good! There are minor things to consider. I'll approve once the tests are passed.

File: src/Common/tendency.jl [hunk 2]
```diff
@@ -150,12 +181,11 @@ end
...
-                old_pdists[i],
-                mom_i[i],
-                param_range = (; :k => (0.1, 10.0)),
-            )
+                old_pdists[i], 
+                mom_i[i], 
+                param_range = (; :k => (1.0, 10.0)))
         else
-            CL.ParticleDistributions.update_dist_from_moments(old_pdists[i], mom_i[i])
+            CL.ParticleDistributions.update_dist_from_moments(old_pdists[i], mom_i[i][1:2])
```
[the remark this answers is not in the document]
Comment (edejong-caltech) [reply src/Common/tendency.jl:188]: Good catch!

File: src/Common/tendency.jl [hunk 3]
```diff
@@ -64,18 +64,51 @@ end
...
-    (; q_tot, q_liq, q_ice) = aux.microph_variables
+    (; moments, pdists, q_rai, N_rai, N_liq, q_tot, q_liq, q_ice) = aux.microph_variables
+    (; tmp_cloudy) = aux.scratch
+    (; weighted_vt) = aux.velocities
+
+    FT = eltype(Y.ρq_vap)
+
+    @. moments = Y.moments
+    @. pdists = get_updated_pdists(moments, pdists, cloudy_params)
+    @. weighted_vt = get_weighted_vt(moments, pdists, cloudy_params)
```
[the remark this answers is not in the document]
Comment (edejong-caltech) [reply src/Common/tendency.jl:99]: Good idea

File: test/experiments/KiD_driver/parse_commandline.jl [hunk 4]
```diff
@@ -76,7 +80,7 @@ function parse_commandline()
         "--t_end"
         help = "Time at the end of the simulation [s]"
         arg_type = Float64
-        default = Float64(3600)
+        default = Float64(10)
```
[the remark this answers is not in the document]
Comment (edejong-caltech) [reply test/experiments/KiD_driver/parse_commandline.jl:83]: No, this was a leftover from testing

File: test/create_parameters.jl (hunk 1 above)
Comment (edejong-caltech) [reply test/create_parameters.jl:88]: This was meant to state whether we should use more exponential modes, versus fewer gamma modes. You're right, I assign the largest mode to be the gamma mode by default (this is typical for cloud/rain), but I'll add an additional kwarg so this is an option.

File: test/experiments/KiD_driver/parse_commandline.jl [hunk 5]
```diff
@@ -15,11 +15,15 @@ function parse_commandline()
...
-        default = "NonEquilibriumMoisture"
+        default = "EquilibriumMoisture"
         "--precipitation_choice"
         help = "Precipitation model choice: NoPrecipitation, Precipitation0M, Precipitation1M, Precipitation2M, CloudyPrecip"
         arg_type = String
         default = "Precipitation1M"
+        "--num_moments"
+        help = "Number of moments to use for CloudyPrecip (ignored otherwise)"
+        arg_type = Int
+        default = 7
```
Comment (trontrytel) [inline test/experiments/KiD_driver/parse_commandline.jl:26]: Should it be 6 by default? Maybe thats why the file does not get created and you get the netcdf error?

**Review by sajjadazimi — APPROVED** [review]

**Pull request merged**